crates/core_simd/src/ops/deref.rs

```diff
@@ -0,0 +1,70 @@
+//! This module hacks in "implicit deref" for Simd's operators.
+//! Ideally, Rust would take care of this itself,
+//! and method calls usually handle the LHS implicitly.
+//! So, we'll manually deref the RHS.
+use super::*;
+
+macro_rules! deref_ops {
+    ($(impl<T, const LANES: usize> $trait:ident<&Self> for Simd<T, LANES> {
+            fn $call:ident(rhs: &Self)
+        })*) => {
+        $(impl<T, const LANES: usize> $trait<&Self> for Simd<T, LANES>
```

Comment:
I think we should also have `impl $trait<Simd<T, LANES>> for &Simd<T, LANES>` and `impl $trait for &Simd<T, LANES>`

Reply:
I would have that be `impl<'a, 'b, T, const LANES: usize> Add<&'a Simd<T, LANES>> for &'b Simd<T, LANES>`, cuz that's more flexible.

Reply:
Does that work when `'b: 'a`?

Reply:
> 

yes, cuz it works for any `'a` and `'b` no matter their relation.